src/main/java/com/flagsmith/flagengine/features/FeatureStateModel.java

```diff
@@ -80,4 +82,14 @@ public boolean equals(Object o) {
 
     return this.getFeature().getId() == ((FeatureStateModel) o).getFeature().getId();
   }
-}
\ No newline at end of file
+
+  public boolean isHigherPriority(FeatureStateModel other) {
+    if (this.featureSegment == null) {
+      return false;
+    } else if (other.featureSegment == null) {
+      return this.featureSegment != null;
```

Comment:
Shouldn't this just be return `true`?

Reply:
It's an interesting question but I'd say no? It feels more reasonable that, if they're both `null`, then neither is a higher priority than the other?

Reply:
No, what I am saying is that isn't `this.featureSegment != null;` is going to be true every time since we have an early exit for `this.featureSegment == null`?

Reply:
No, what I am saying isn't `this.featureSegment != null` another way of writing `true`? Since whenever this line will be executed, it will always be true?

Reply:
I've tidied this logic up a little bit and added some documentation (as it got a bit confusing).